### PR TITLE
schemacmp: add compatibility for integer and blob

### PR DESCRIPTION
**Pull request by csuzhangxc**
### What problem does this PR solve? 

add compatibility compare for MySQL integer types and blob types.

### What is changed and how it works?

compare different MySQL integer types and blob types.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change

Side effects

 - Possible performance regression
 - Increased code complexity

### CODE REVIEW

**Comment by csuzhangxc** [comment]
@WangXiangUSTC @lichunzhu PTAL

**Review by kennytm — COMMENTED** [review]
LGTM

File: pkg/schemacmp/lattice.go [hunk 1]
```diff
@@ -278,6 +289,136 @@ func (a Byte) Join(other Lattice) (Lattice, error) {
...
+		case b.value == mysql.TypeInt24:
+			if a.value <= mysql.TypeShort {
+				return -1, nil
+			}
+			return 1, nil
+		case a.value < b.value:
+			return -1, nil
+		default:
+			return 1, nil
+		}
```
Comment (WangXiangUSTC) [inline pkg/schemacmp/lattice.go]: can we define a function to compare Integer type? and this function can be reused in `Join`

File: pkg/schemacmp/lattice.go [hunk 2]
```diff
@@ -278,6 +289,136 @@ func (a Byte) Join(other Lattice) (Lattice, error) {
...
+		case b.value == mysql.TypeBlob:
+			if a.value == mysql.TypeTinyBlob {
+				return -1, nil
+			}
+			return 1, nil
+		case a.value < b.value:
+			return -1, nil
+		default:
+			return 1, nil
+		}
```
Comment (WangXiangUSTC) [inline pkg/schemacmp/lattice.go]: ditto

**Review by WangXiangUSTC — COMMENTED** [review]
rest LGTM

**Review by lichunzhu — COMMENTED** [review]
LGTM

File: pkg/schemacmp/lattice.go (hunk 1 above)
Comment (csuzhangxc) [reply pkg/schemacmp/lattice.go]: addressed in https://github.com/pingcap/tidb-tools/pull/338/commits/fc42099d797032a437a22ce9f614f30af98a8106.

File: pkg/schemacmp/lattice.go (hunk 2 above)
Comment (csuzhangxc) [reply pkg/schemacmp/lattice.go]: addressed in https://github.com/pingcap/tidb-tools/pull/338/commits/fc42099d797032a437a22ce9f614f30af98a8106.

**Review by lichunzhu — APPROVED** [review]
LGTM

**Review by WangXiangUSTC — APPROVED** [review]
LGTM

**Pull request merged**